src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key] as string;
+
+    const newResult = { ...result };
+
+    if (!newResult[keyValue]) {
+      newResult[keyValue] = [];
+    }
+
+    newResult[keyValue].push(item);
+
+    return newResult;
+  }, {} as GroupsMap<T>);
```

Comment:
The initial value for the reduce function `{}` should be typed as `GroupsMap<T>` to avoid the type assertion `as GroupsMap<T>`. You can define the initial value outside of the reduce call and use it directly.